azurelinuxagent/agent.py

```diff
@@ -74,15 +123,24 @@ def main():
     elif command == "start":
         start()
     else:
-        agent = Agent(verbose)
-        if command == "deprovision+user":
-            agent.deprovision(force, deluser=True)
-        elif command == "deprovision":
-            agent.deprovision(force, deluser=False)
-        elif command == "register-service":
-            agent.register_service()
-        elif command == "daemon":
-            agent.daemon()
+        try:
+            agent = Agent(verbose)
+            if command == "deprovision+user":
+                agent.deprovision(force, deluser=True)
+            elif command == "provision":
+                agent.provision()
+            elif command == "deprovision":
+                agent.deprovision(force, deluser=False)
+            elif command == "register-service":
+                agent.register_service()
+            elif command == "daemon":
+                agent.daemon()
+            elif command == "update":
```

Comment:
is `waagent -update` the new entrypoint if we want to have self-update + GA functionality?